### ISSUE TITLE
Warn when weaviate and/or client are outdated

### ISSUE THREAD

**Issue by dirkkul** [weaviate/typescript-client#41]
Some users use very old versions of weaviate and client. We should warn them if it the minor version is 3 or more versions behind the last release.

Eg currently the last Weaviate version is 1.18.3, so we would warn users of 1.15 or below. We could do the same for the client versions.

**Comment by Abhijithchintu** [comment]
Hi @dirkkul . Is this issue still open. If it is open, can i work on this issue ?

**Comment by dirkkul** [comment]
Hi, yes this is still open! 

Here are the python PRs that implemeted that functionality:
https://github.com/weaviate/weaviate-python-client/pull/343
https://github.com/weaviate/weaviate-python-client/pull/299

**Comment by Abhijithchintu** [comment]
Please review  
https://github.com/weaviate/typescript-client/pull/81

**Comment by tsmith023** [comment]
@Abhijithchintu, thanks for the contribution! https://github.com/weaviate/typescript-client/pull/81 is now merged, which covers warning for outdated Weaviate. Warning for an outdated client is still open, would you like to open a PR for that also?

**Comment by Abhijithchintu** [comment]
Yes , i will work on warning for an outdated client. I have few doubts in that.

1. how to fetch typescript client version 
2. Should we show the warning for client version (if applicable ) at the same time as weaviate version warning.

**Comment by tsmith023** [comment]
In the Python client, the version of the package is obtained [here](https://github.com/weaviate/weaviate-python-client/blob/main/weaviate/__init__.py#L61), this is then used [here](https://github.com/weaviate/weaviate-python-client/blob/main/weaviate/connect/connection.py#L672) when instantiating a `Connection` object alongside the check for the Weaviate server version.

I would say we follow a similar logic for the TypeScript client so warn of outdated client in the same place as warning for outdated server. For retrieving the version, there are concerns in TypeScript around revealing the entire `package.json` file to the browser when this client is used in frontend apps. As such, the recommendation [here](https://stackoverflow.com/a/67701490) should be followed.

This `prebuild` script will be used within other build, test, and package scripts to ensure the version is always up-to-date without compromising the security of the `package.json` file.

**Comment by eskayML** [comment]
i'll take the client side of this. the server warning shipped in #81, this half never did.

checked current main. WEAVIATE_CLIENT_VERSION is written by scripts/gen-version.mjs into src/version.ts but it's only used for the X-Weaviate-Client header. nothing checks if the client itself is behind.

plan is to pull the latest version from npm once per process on connect and warn when we sit 3+ minor versions behind. silent on failure, offline or ci. node only though. runtime registry calls in a browser bundle are the reason this stalled, i don't want to add that without a proper discussion.

want me to open the pr scoped to node?
